tiled/catalog/adapter.py

```diff
@@ -658,22 +658,19 @@ async def create_node(
                 node.data_sources.append(data_source_orm)
                 await db.flush()  # Get data_source_orm.id.
                 for asset in data_source.assets:
-                    try:
-                        # Find an asset if it exists
-                        statement = select(orm.Asset.id).where(
-                            orm.Asset.data_uri == asset.data_uri
+                    # Find an asset_id if it exists, otherwise create a new one
+                    statement = select(orm.Asset.id).where(
+                        orm.Asset.data_uri == asset.data_uri
+                    )
+                    result = await db.execute(statement)
+                    if result := result.fetchone():
```

Comment:
Great use the "walrus operator". Minor nit: can we use a new variable name (maybe `row`?) instead of shadowing `result`?

Reply:
done!